### ISSUE TITLE
Error when running without GPU

### ISSUE THREAD

**Issue by warren-oneill** [jcjohnson/neural-style#3]
I'm trying to run the the following on a VM without a graphics card
```
th neural_style.lua -style_image image1.jpg -content_image image2.jpg -gpu -1
```
and I get the following error
```
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:505] Reading dangerously large protocol message.  If the message turns out to be larger than 1073741824 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:78] The total number of bytes read was 574671192
Successfully loaded models/VGG_ILSVRC_19_layers.caffemodel
modprobe: FATAL: Module nvidia not found.
/home/dev/torch/install/bin/luajit: /home/dev/torch/install/share/lua/5.1/trepl/init.lua:363: /home/dev/torch/install/share/lua/5.1/trepl/init.lua:363: cuda runtime error (38) : no CUDA-capable device is detected at /tmp/luarocks_cutorch-scm-1-3249/cutorch/lib/THC/THCGeneral.c:16
stack traceback:
        [C]: in function 'error'
        /home/dev/torch/install/share/lua/5.1/trepl/init.lua:363: in function 'require'
        models/VGG_ILSVRC_19_layers_deploy.prototxt.lua:2: in main chunk
        [C]: in function 'dofile'
        .../dev/torch/install/share/lua/5.1/loadcaffe/loadcaffe.lua:20: in function 'load'
        neural_style.lua:47: in function 'main'
        neural_style.lua:282: in main chunk
        [C]: in function 'dofile'
        .../dev/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
        [C]: at 0x00405ea0

```
any tips?

**Comment by jcjohnson** [comment]
In your models directory you should have a file named `VGG_ILSVRC_19_layers_deploy.prototxt.lua` - can you post it?

**Comment by jcjohnson** [comment]
Looks like the problem is with loadcaffe right here:

https://github.com/szagoruyko/loadcaffe/blob/master/loadcaffe.cpp#L102

every time you run loadcaffe, it converts the prototxt file into a .lua file that builds the model described by the prototxt, then executes the file to actually build the model. The crash you are seeing is when the generated file is executed, specifically when it tries to import cunn.

The problem is that the file generated by loadcaffe **always** import cunn, which will crash if you only want to run in CPU mode. I will see if there is a quick workaround.

**Comment by jcjohnson** [comment]
Looks like loadcaffe is aware of the problem; they already have an open issue here:

https://github.com/szagoruyko/loadcaffe/issues/8

In the meantime, I added a **very** quick and dirty fix wit this commit:

https://github.com/jcjohnson/neural-style/commit/bfa24329cdbc8f6e0512e6a07f9ad9bcdd3638f8

Let me know if this fixes the problem for you.

**Comment by warren-oneill** [comment]
it works!! here's what I get when I combine my avatar with blue marble
![out](https://cloud.githubusercontent.com/assets/10514666/9634552/8c4fded4-5192-11e5-8958-0722d851463c.png)

**Comment by warren-oneill** [comment]
hey @jcjohnson is this output normal?
```
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:505] Reading dangerously large protocol message.  If the message turns out to be larger than 1073741824 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:78] The total number of bytes read was 574671192
Successfully loaded models/VGG_ILSVRC_19_layers.caffemodel
conv1_1: 64 3 3 3
conv1_2: 64 64 3 3
conv2_1: 128 64 3 3
conv2_2: 128 128 3 3
conv3_1: 256 128 3 3
conv3_2: 256 256 3 3
conv3_3: 256 256 3 3
conv3_4: 256 256 3 3
conv4_1: 512 256 3 3
conv4_2: 512 512 3 3
conv4_3: 512 512 3 3
conv4_4: 512 512 3 3
conv5_1: 512 512 3 3
conv5_2: 512 512 3 3
conv5_3: 512 512 3 3
conv5_4: 512 512 3 3
fc6: 1 1 25088 4096
fc7: 1 1 4096 4096
fc8: 1 1 4096 1000
WARNING: Skipping content loss
Iteration 10 / 1000
  Content 1 loss: 367002.062500
  Style 1 loss: 18768.949890
  Style 2 loss: 441930.078125
  Style 3 loss: 106142.565918
  Style 4 loss: 1903146.875000
  Style 5 loss: 4086091.406250      
```

**Comment by jcjohnson** [comment]
Looks good to me!

**Comment by warren-oneill** [comment]
:smile:

**Comment by chirag64** [comment]
Hi @warren-oneill, can you please share what style image you used on your avatar? The output image looks awesome. By 'blue marble', I was guessing you meant the 'blue marble photograph' - https://en.wikipedia.org/wiki/The_Blue_Marble#/media/File:The_Earth_seen_from_Apollo_17.jpg, but that gives a much more blackish output than the blue one you have here.

**Comment by warren-oneill** [comment]
hey I meant this kind of blue marble :smile: http://www.burramundi.com/wp-content/uploads/2010/08/Electric-Blue-Marble_1920x1200.jpg

**Comment by chirag64** [comment]
I ended up using your output image as style image instead. That gave interesting results too. :smile:
